gs_quant/timeseries/arima.py

```diff
@@ -0,0 +1,246 @@
+
+
+"""ARIMA is the Autoregressive Integrated Moving Average Model and is used 
+to normalize and forecast time series data. ARIMA has 3 parameters: (p, d, q)
+where:
+    :p is the number of autoregressive terms
+    :d is the number of nonseasonal differences
+    :q is the number of lagged forecast errors in the prediction equation
+    
+An ARIMA model is selected from the Catesian product of sets p, q, and d. The 
+time series is split into train and test sets and an ARIMA model is fit for 
+every combination on the training set. The model with the lowest mean-squared 
+error (MSE) on the test set is selected as the best model. The original 
+times series can then be transformed by the best model.
+
+Autoregressive components are past values of the variable of interest. An 
+AR(p) model with order p = 1 may be written as X(t) = A(1) * X(t-1) + E(t),
+where
+    :X(t) is the time series under investigation 
+    :A(1) is the autoregressive parameter
+    :X(t-1) is the time series lagged 1 period
+    :E(t) is the error term of the model or white noise
+
+In other words, any value in X(t) can be explained using a linear 
+combination of the past value T(t-1) plus some error term E(t). X(t)
+could also be a linear combination of more than one past value: 
+X(t) = A(1) * X(t-1) + A(2) * X(t-2) + E(t).
+
+Differencing is a way of making a non-stationary time series stationary. This 
+is done by computing the differences between consuective observations 
+(subtracting the observation from the current period from the previous one).
+Differencing can help stabilize the mean of a time series by removing changes 
+in the level of a time series, which reduces trend and seasonality. If the 
+transformation is done once, then the data has been "first differenced". The 
+same transformation can be done again, so the data would be "second 
+differenced".
+
+Moving average components uses past forecast errors E(t). In other words,
+X(t) can be thought of as a weighted moving average of the past forecast 
+errors: X(t) = c + E(t) + W(1)*E(t-1) + ... + W(q)*E(t-q).
+
+"""
+
+
+class arima():
+    """
+    An ARIMA class used to normalize time series data.
+    """
+
+
+    def __init__(self):
+        self.best_params = {}
+    
+
+    def _evaluate_arima_model(self, X: Union[pd.Series, pd.DataFrame], arima_order: Tuple[int, int, int], train_size: float, freq: str) -> Tuple[float, dict]:
+        train_size = int(len(X) * train_size)
+        train, test = X[0:train_size].astype(float), X[train_size:].astype(float)
+
+        model = ARIMA(train, order=arima_order, freq=freq)
+        model_fit = model.fit(disp=0)
+        ar_coef = model_fit.arparams
+        ma_coef = model_fit.maparams
+        resid = model_fit.resid
+
+        model_params = model_fit.params.to_dict()
+        const = model_params.get("const", 0)
+
+        # calculate test error
+        yhat = model_fit.forecast(len(test))[0]
+        error = mean_squared_error(test, yhat)
```

Comment:
this func introduces skikit learn dependency? could you check that ARIMA doesn't have anything from MSE? We would want this function exposed for the timeseries module as well